### ISSUE TITLE
What models, optimizers, losses and activations are available?

### ISSUE THREAD

**Issue by codeananda** [AIStream-Peelout/flow-forecast#358]
There is no clear guidance in the documentation about which models, optimizers, losses, or activations are available to use.

I had to dig through the source code to find them. We should definitely put them in the documentation and explain what each of them do, when you might use them, and what you need to change if you use one over the other.

I've put down questions underneath that went through my mind yesterday as I read the docs/source code. A clear explanation for each would be wonderful.

- `activation_dict` in lower_upper_config.py, contains all available activations:s ReLu, Softmax, etc. 
  - Used in MultiAttnHeadSimple class to choose the final layer activation (though this is not documented) and the default value is False with no explanation. Do I need to manually pick a different activation based on my problem? Does FF handle this automatically? What should I do for (multi-class)classification/regression.
- `pytorch_model_dict` in model_dict_function.py - list of available models
  - What inputs do they need? Do they all take the same inputs? How to use them for classification/regression?
- `pytorch_criterion_dict` in model_dict_function.py - list of available losses
  - When to use which loss? Which are for classification and which are for regression?
  - If I use NegativeLogLikelihood, do I need to use a particular final layer? If I code my own PyTorch module and want to use NLLLoss(), I must use F.log_softmax() on the final layer, not ordinary softmax. But Log_SoftMax is not an available final layer. Does this mean I cannot use FF for multi-class classification?
- `pytorch_opt_dict` in model_dict_function.py - list of optimizers
  - What params do they take?
  - Make it clear that Adam and SGD are from torch and not custom built.

A quick fix would be to add a short section to the docs telling readers to go to the above files to look at the available options. Currently we are completely in the dark.
